pageserver/src/tenant/timeline/delete.rs

```diff
@@ -38,6 +38,14 @@ async fn stop_tasks(timeline: &Timeline) -> Result<(), DeleteTimelineError> {
     }
     debug!("wal receiver shutdown confirmed");
 
+    // Shut down the layer flush task before the remote client, as one depends on the other
```

Comment:
What is this dependency?

Reply:
Flushing layers to remote storage depends on the remote timeline client's queue being available -- the current symptom is that we get an error logged when it tries to flush a layer with the queue in Stopping state.